lint/analyzer.go

```diff
@@ -31,6 +31,7 @@ const (
 	UpdateCategory          = "update recommended"
 	UnnecessaryCastCategory = "unnecessary-cast-hint"
 	DeprecatedCategory      = "deprecated"
+	CadenceV1Category       = "cadence-v1"
```

Comment:
I wasn't entirely sure what category to use here.  It seems like some of the categories define what type of code edit to apply (e.g. `ReplacementCategory`) and some of the categories define the the type of diagnostic they provide (e.g. `UnnecessaryCastCategory`).  I'm not sure if this pre-dates the existence of `SuggestedFix.TextEdits` or something but I have just set all the C1.0 diagnostics to `CadenceV1Category` for now and can change if needed